lucille-core/src/main/java/com/kmwllc/lucille/core/Indexer.java

```diff
@@ -71,6 +73,14 @@ public Indexer(Config config, IndexerMessageManager manager, String metricsPrefi
         config.hasPath("indexer.deletionMarkerFieldValue")
             ? config.getString("indexer.deletionMarkerFieldValue")
             : null;
+    this.deleteByFieldField =
```

Comment:
Should enforce that when one of deleteByFieldField and deleteByFieldValue are present, both are present
Similar with deletionMarkerFieldField and deletionMarkerFieldValue

Runner should be able to handle an IllegalArgumentException thrown here but we'd need to confirm & test
Could also update Indexer constructor to throw IndexerException
Longer term, we need a way to validate Indexer and Worker configs like we validate Stage configs

OK to leave this out of the current PR but need a ticket to revisit

Reply:
I tried an experiment locally, throwing an unchecked exception from Indexer() and the run did not terminate gracefully.
We need to add a try catch around [this line in Runner](https://github.com/kmwtechnology/lucille/blob/34010ac3a988c8c23278986efc459409d2f5db5d/lucille-core/src/main/java/com/kmwllc/lucille/core/Runner.java#L395). 
It should log and error and return in ConnectorResult in the same way as the code directly below it.
We'd want to have a test in RunnerTest that invokes that error handling logic in some way.
I can help with those Runner and RunnerTest changes in a separate PR but if you'd like to do them here that's good too.